aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -304,12 +306,19 @@ public void onError(Exception error) {
 
     @Override
     public void confirmSignUp(
-        @NonNull String username,
-        @NonNull String confirmationCode,
-        @NonNull final Consumer<AuthSignUpResult> onSuccess,
-        @NonNull final Consumer<AuthException> onException
+            @NonNull String username,
+            @NonNull String confirmationCode,
+            @NonNull AuthConfirmSignUpOptions options,
+            @NonNull final Consumer<AuthSignUpResult> onSuccess,
+            @NonNull final Consumer<AuthException> onException
     ) {
-        awsMobileClient.confirmSignUp(username, confirmationCode, new Callback<SignUpResult>() {
+        final Map<String, String> clientMetadata = new HashMap<>();
+
+        if (options instanceof AWSCognitoAuthConfirmSignUpOptions) {
+            clientMetadata.putAll(((AWSCognitoAuthConfirmSignUpOptions) options).getClientMetadata());
```

Comment:
nit: just for readability

```suggestion
            AWSCognitoAuthConfirmSignUpOptions cognitoOptions = (AWSCognitoAuthConfirmSignUpOptions) options;
            clientMetadata.putAll(cognitoOptions.getClientMetadata());
```